.github/workflows/bump-versions.yaml

```diff
@@ -0,0 +1,112 @@
+          for commit in ${{ github.event.push.commits }}; do
+            if [[ "${commit.message}" == *"Update-versions "* ]]; then
+              # Get outputs in form of chart:major/minor/patch app:major/minor/patch
+              outputs=$( echo "${commit.message}" | sed -rn "s/Update-versions (app|chart) (\w+)( (app|chart) (\w+))?/\1:\2 \4:\5/p" | sed "s/ ://" )
+              # Support the default case which is patch chart version
+              if [[ "$outputs" != *"chart:" ]]; then
+                  outputs="$outputs chart:patch"
+              fi
+              echo( "Using $outputs versions from commit message ${commit.message}" )
+              break
+            fi
+          done
+          
+          # Check both chart and app for version changes and bump accordingly 
+          vers=( chart app )
+          for i in "${vers[@]}"; do
+              if [ "$i" = "chart" ]; then
+                  version=$CHARTVERSION
+              else
+                  version=$APPVERSION
+              fi
+              rNumRegex='^r[[:digit:]]+$'
+              semRegex='^[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+$'
+              # If versioning follows pattern r123 where 123 is any set of digits
+              if [[ $version =~ $rNumRegex ]]; then
+                  echo "Version fits pattern r123 where 123 are any digits"
+                  # If the versioning is r123 format, ignore patch/major/minor as it can only be a +1
+                  echo "outputs: ${outputs}"
+                  if [[ "${outputs}" == *"${i}:"* ]]; then
+                      newVers="${version:1}"
+                      ((newVers++))
+                      printf -v version 'r%d' "$((newVers))"
+                  fi
+              # If versioning follows semantic versioning pattern like 0.12.23
+              elif [[ $version =~ $semRegex ]]; then
+                  echo "Version uses semantic versioning"
+                  echo "original: $version"
+                  IFS=. read -r major minor patch <<<"$version"
+                  if [[ "${outputs}" == *"${i}:patch"* ]]; then
+                      ((patch++))
+                  elif [[ "${outputs}" == *"${i}:minor"* ]]; then
+                      ((minor++))
+                      patch="0"
+                  elif [[ "${outputs}" == *"${i}:major"* ]]; then
+                      ((major++))
+                      patch="0"
+                      minor="0"
+                  fi
+                  printf -v version '%d.%d.%d' "$((major))" "$((minor))" "$((patch))"
+              else echo "${version} did not match regex pattern for 1.2.3 or r123"
+              fi
+
+              if [ "$i" = "chart" ]; then
+                  echo "new chart vers: ${version}"
+                  echo "::set-output name=newChartversion::${version}"
+              else
+                  echo "new app vers: ${version}"
+                  echo "::set-output name=newAppversion::${version}"
+              fi
+          done
```

Comment:
nit: Looks like 4-space indentation is being used here, while 2-space is being used elsewhere. Would prefer 2-space throughout, as per https://backstage.niche.team/docs/default/resource/sre-team/best-practices/

Reply:
That makes sense! A lot of it was copied from a local bash script where I tested so I missed that, will fix